libsolidity/formal/SMTEncoder.cpp

```diff
@@ -348,13 +348,46 @@ void SMTEncoder::endVisit(VariableDeclarationStatement const& _varDecl)
 
 }
 
+bool SMTEncoder::visit(Assignment const& _assignment)
+{
```

Comment:
I had this block of code in `endVisit(Assignment)`, but it doesn't work for `a.push() = a.push()` because of the visit order, so had to move it here. Still not 100% sure, but couldn't come up with anything better so far

Reply:
If there is no better way, let's leave it as is